### PR TITLE
Feature/center recipe modal and fix ingredient amounts

### PR DESCRIPTION
**Pull request by mbenfowler**
## Description
This pr:
- Moves the recipe modal to the center of the page and gives it a `max-width`
  - user can no longer scroll the main recipe list while viewing a recipe in the modal
- TDD for a function to remove extra decimals past 2 on ingredient amounts

Testing Notes
- Someone with a big screen, please test that the recipe modal is actually in the center for I cannot >_<
- Click on the "Pastry Cream" recipe at the bottom of the page
  - Ensure sucrose only has 2 decimal places
  - No other ingredients that have 2 or less decimal places should be affected

### CODE REVIEW

**Comment by mbenfowler** [comment]
[[refactoring] center recipe card](https://trello.com/c/ZlFywsHZ/23-refactoring-center-recipe-card)

File: src/styles.css [hunk 1]
```diff
@@ -376,9 +376,11 @@ h3 {
   position: fixed;
   justify-content: center;
   align-items: center;
-  top: 0;
-  right: 0;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
   width: 98%;
+  max-width: 1200px;
```
Comment (taranveersingh93) [inline src/styles.css]: Can we achieve the same visuals if we change width from 98% to 60%?

File: src/styles.css [hunk 2]
```diff
@@ -376,9 +376,11 @@ h3 {
   position: fixed;
   justify-content: center;
   align-items: center;
-  top: 0;
-  right: 0;
+  left: 50%;
+  top: 50%;
```
Comment (taranveersingh93) [inline src/styles.css:371]: Also, if we're going for reduced width: how do you feel about reduced height as well? in 75% or 80% range? If it looks good to you.

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]

File: src/styles.css (hunk 1 above)
Comment (lauraguerra1) [reply src/styles.css]: @taranveersingh93 I feel like 60% is very small. Perhaps width at 85% would keep it dependent on the screen size without having to use pixels, and that would make it smaller, if we want it smaller. I tested out width at 60% and I think that looks too small.

File: src/domUpdates.js [hunk 3]
```diff
@@ -255,6 +256,7 @@ const populateRecipeHeader = currentRecipe => {
 
 const openRecipeCard = () => {
   allRecipes.classList.add('blur')
+  body.classList.add('no-scroll')
```
Comment (lauraguerra1) [inline src/domUpdates.js:237]: Nice!

File: src/recipes.js [hunk 4]
```diff
@@ -21,13 +21,18 @@ const getIngredients = (recipe, ingredients) => {
...
       unit: ingredient.quantity.unit,
       name: getIngredientProperty(ingredient, ingredients, 'name')
     }
   })
 }
 
+const fixIngredientAmount = (amount) => {
+  let remainder = amount % 1;
+  return remainder.toString().length > 2 ? Number(amount.toFixed(2)) : amount;
+}
```
Comment (lauraguerra1) [inline src/recipes.js:33]: love this

**Review by lauraguerra1 — APPROVED** [review]

**Review by taranveersingh93 — APPROVED** [review]
Thank you!

**Pull request merged**